### PR TITLE
Refactor logic

### PR DESCRIPTION
**Pull request by lamgj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,9 +3,9 @@
 Latin for *work*.
 
 [![Build Status](https://travis-ci.org/CS2103JAN2017-W15-B3/main.svg?branch=master)](https://travis-ci.org/CS2103JAN2017-W15-B3/main)
-[![Build status](https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true)](https://ci.appveyor.com/project/damithc/addressbook-level4)
-[![Coverage Status](https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master)](https://coveralls.io/github/se-edu/addressbook-level4?branch=master)
-[![Codacy Badge](https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a)](https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=se-edu/addressbook-level4&amp;utm_campaign=Badge_Grade)
+[![Build status](https://ci.appveyor.com/api/projects/status/12mctlo4ld7a7aqi?svg=true)](https://ci.appveyor.com/project/xbili/main)
```
Comment (lamgj) [inline README.md:6]: I have no idea why my commits will include these changes when I didn't stage them :/
Can someone resolve this?

File: _config.yml [hunk 2]
```diff
@@ -1 +1 @@
-theme: jekyll-theme-cayman
+theme: jekyll-theme-midnight
```
Comment (lamgj) [inline _config.yml:1]: Same goes for this

**Comment by lamgj** [comment]
Checks are still failing as I require some of logic components depend on model and one on ui. So will check again after all the refactoring commits are done after merging into development tonight

File: README.md (hunk 1 above)
Comment (mynameisyz) [reply README.md:6]: This file change is done commit 5ef8dd4 and 3fbfa21. But I'm not sure why they are in this branch though

File: _config.yml (hunk 2 above)
Comment (mynameisyz) [reply _config.yml:1]: File change in commit a4daf9b

File: README.md (hunk 1 above)
Comment (mynameisyz) [reply README.md:6]: I have referenced the wrong commits. But they are done in earlier commits in this branch.

Comment (xbili) [reply README.md:6]: No worries, there won't be a merge conflict when we eventually merge in.

**Comment by xbili** [comment]
@gjlam95 I'll revert some of the changes made to the test cases.

Side note: Branches should always be passing when we merge. We can make the changes to the test cases again in a separate branch.

**Review by xbili — APPROVED** [review]

**Pull request merged**